src/components/MultiHopTrade/components/TradeInput/components/HorizontalCollapse.tsx

```diff
@@ -0,0 +1,51 @@
+import { Box } from '@chakra-ui/react'
+import type { MotionStyle } from 'framer-motion'
+import { motion } from 'framer-motion'
+import { useCallback, useMemo, useState } from 'react'
+
+export type HorizontalCollapseProps = {
+  isOpen: boolean
+  children: JSX.Element
+  width: string | number
+  height: string | number
+}
+
+export const HorizontalCollapse = ({
+  isOpen,
+  children,
+  width,
+  height,
+}: HorizontalCollapseProps) => {
+  const [hidden, setHidden] = useState(!isOpen)
+
+  const motionStyle = useMemo(
+    () =>
+      ({
+        height,
+        overflow: 'hidden',
+        whiteSpace: 'nowrap',
+        position: 'relative',
+      }) as MotionStyle,
+    [height],
+  )
```

Comment:
For type safety, declare the `MotionStyle` type, rather than asserting it:

```suggestion
const motionStyle = useMemo(
  (): MotionStyle => ({
    height,
    overflow: 'hidden',
    whiteSpace: 'nowrap',
    position: 'relative',
  }),
  [height],
)
```

Reply:
nice, done!